go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -0,0 +1,605 @@
+		} else {
+			v4Gateway = gwIfAddr.IP
+			excludeIPs := mgmtIfAddr.IP.String()
+			if config.HybridOverlay.Enabled {
+				hybridOverlayIfAddr := util.GetNodeHybridOverlayIfAddr(hostSubnet)
+				excludeIPs += ".." + hybridOverlayIfAddr.IP.String()
+			}
+			logicalSwitch.OtherConfig["subnet"] = hostSubnet.String()
+			logicalSwitch.OtherConfig["exclude_ips"] = excludeIPs
+		}
+	}
+
+	if loadBalancerGroupUUID != "" {
+		logicalSwitch.LoadBalancerGroup = []string{loadBalancerGroupUUID}
+	}
+
+	logicalRouterPortName := types.RouterToSwitchPrefix + switchName
+	logicalRouterPort := nbdb.LogicalRouterPort{
+		Name:     logicalRouterPortName,
+		MAC:      nodeLRPMAC.String(),
+		Networks: logicalRouterPortNetwork,
+	}
+	logicalRouterName := types.OVNClusterRouter
+	logicalRouter := nbdb.LogicalRouter{Name: logicalRouterName}
+
+	err := libovsdbops.CreateOrUpdateLogicalRouterPorts(bnc.nbClient, &logicalRouter,
+		[]*nbdb.LogicalRouterPort{&logicalRouterPort}, &logicalRouterPort.Networks, &logicalRouterPort.MAC)
+	if err != nil {
+		return fmt.Errorf("failed to add logical router port %+v to router %s: %v", logicalRouterPort, logicalRouterName, err)
+	}
+
+	// If supported, enable IGMP/MLD snooping and querier on the node.
+	if bnc.multicastSupport {
+		logicalSwitch.OtherConfig["mcast_snoop"] = "true"
+
+		// Configure IGMP/MLD querier if the gateway IP address is known.
+		// Otherwise disable it.
+		if v4Gateway != nil || v6Gateway != nil {
+			logicalSwitch.OtherConfig["mcast_querier"] = "true"
+			logicalSwitch.OtherConfig["mcast_eth_src"] = nodeLRPMAC.String()
+			if v4Gateway != nil {
+				logicalSwitch.OtherConfig["mcast_ip4_src"] = v4Gateway.String()
+			}
+			if v6Gateway != nil {
+				logicalSwitch.OtherConfig["mcast_ip6_src"] = util.HWAddrToIPv6LLA(nodeLRPMAC).String()
+			}
+		} else {
+			logicalSwitch.OtherConfig["mcast_querier"] = "false"
+		}
+	}
+
+	err = libovsdbops.CreateOrUpdateLogicalSwitch(bnc.nbClient, &logicalSwitch, &logicalSwitch.OtherConfig,
+		&logicalSwitch.LoadBalancerGroup)
+	if err != nil {
+		return fmt.Errorf("failed to add logical switch %+v: %v", logicalSwitch, err)
+	}
+
+	// Connect the switch to the router.
+	logicalSwitchPort := nbdb.LogicalSwitchPort{
+		Name:      types.SwitchToRouterPrefix + switchName,
+		Type:      "router",
+		Addresses: []string{"router"},
+		Options:   map[string]string{"router-port": types.RouterToSwitchPrefix + switchName},
+	}
+	sw := nbdb.LogicalSwitch{Name: switchName}
+	err = libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitch(bnc.nbClient, &sw, &logicalSwitchPort)
+	if err != nil {
+		klog.Errorf("Failed to add logical port %+v to switch %s: %v", logicalSwitchPort, switchName, err)
+		return err
+	}
+
+	// multicast is only supported in default network for now
+	if bnc.multicastSupport {
+		err = libovsdbops.AddPortsToPortGroup(bnc.nbClient, types.ClusterRtrPortGroupName, logicalSwitchPort.UUID)
+		if err != nil {
+			klog.Errorf(err.Error())
+			return err
+		}
+	}
```

Comment:
Previously we would add the port to the port group regardless of multicast support. I guess doing only in case of multicast support is fine. Just curious if there was a specific reason for changing it.

Reply:
because for secondary networks, we didn't even create its own ClusterRtrPortGroupName port group.